### PR TITLE
sci-electronics/gnucap: Version bump to gnucap-0.36.20171003

### PR DESCRIPTION
**Pull request by Peter-Levine**
Signed-off-by: Peter Levine <plevine457@gmail.com>
Bug: https://bugs.gentoo.org/614656
Package-Manager: Portage-2.3.49, Repoman-2.3.10

### CODE REVIEW

File: sci-electronics/gnucap/gnucap-0.36.20171003.ebuild [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+SNAPSHOTDATE="${P##*.}"
+MY_P="${PN}-${SNAPSHOTDATE}"
+
+DESCRIPTION="GNUCap is the GNU Circuit Analysis Package"
+SRC_URI="http://git.savannah.gnu.org/cgit/${PN}.git/snapshot/${MY_P}.tar.gz"
+HOMEPAGE="http://www.gnucap.org/"
+
+IUSE="+models +readline"
+SLOT="0"
+LICENSE="GPL-2"
```
Comment (felix-salfelder) [inline sci-electronics/gnucap/gnucap-0.36.20171003.ebuild]: gnucap is GPL-3+. thanks

Comment (Peter-Levine) [reply sci-electronics/gnucap/gnucap-0.36.20171003.ebuild]: Ah, I missed that one. OK.

**Comment by Peter-Levine** [comment]
Note: The version was incremented from 0.35 to 0.36 since Debian/Ubuntu lists it as such.  I would assume it is but there is no Changelog entry or make/configure variable to suggest one way or another.

**Comment by felix-salfelder** [comment]
since the switch to the tag "releases" [1], i'm pretty confident that the "0.3{5,6}" versioning has been discontinued. i think what you did is reasonable and good enough

thanks

[1] https://www.mail-archive.com/gnucap-devel@gnu.org/msg01153.html

File: sci-electronics/gnucap-models/gnucap-models-0.36.20171003.ebuild [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit toolchain-funcs
+
+SNAPSHOTDATE="${P##*.}"
```
Comment (mgorny) [inline sci-electronics/gnucap-models/gnucap-models-0.36.20171003.ebuild]: Why P and not PV?

File: sci-electronics/gnucap-models/gnucap-models-0.36.20171003.ebuild [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+DEPEND="~sci-electronics/gnucap-${PV}"
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/${MY_P}"
+
+src_prepare() {
+	default
+
+	sed -i -e 's:INSTALL_DIR = :INSTALL_DIR = ${D}:' \
```
Comment (mgorny) [inline sci-electronics/gnucap-models/gnucap-models-0.36.20171003.ebuild]: What's the D here? You aren't relying on ebuild exporting `D`, are you?

File: sci-electronics/gnucap-models/gnucap-models-0.36.20171003.ebuild [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+		-e 's:C_CC_FLAGS = -O2 -g:C_CC_FLAGS = :' \
+		-e 's:CFLAGS =:CFLAGS +=:' \
+		-e 's:CCFLAGS =:CCFLAGS = ${CXXFLAGS}:' \
+		-e 's:LDFLAGS =:LDFLAGS +=:' \
+		plugins/models-*/Make2 || die "sed failed"
+}
+
+src_compile () {
+	emake CC=$(tc-getCC) CCC=$(tc-getCXX)
```
Comment (mgorny) [inline sci-electronics/gnucap-models/gnucap-models-0.36.20171003.ebuild]: Quoting.

File: sci-electronics/gnucap/gnucap-0.36.20171003.ebuild [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+DESCRIPTION="GNUCap is the GNU Circuit Analysis Package"
+SRC_URI="http://git.savannah.gnu.org/cgit/${PN}.git/snapshot/${MY_P}.tar.gz"
+HOMEPAGE="http://www.gnucap.org/"
+
+IUSE="+models +readline"
+SLOT="0"
+LICENSE="GPL-3+"
+KEYWORDS="~amd64 ~ppc ~x86"
+
+DEPEND="readline? ( sys-libs/readline:= )"
```
Comment (mgorny) [inline sci-electronics/gnucap/gnucap-0.36.20171003.ebuild]: Are non-`0` slots useful?

**Comment by mgorny** [comment]
Wait a minute! I knew something was wrong with this but couldn't grasp what exactly. Please split it into one commit per package.

**Comment by mgorny** [comment]
Oh, and as a side note, traditionally we used `_p<DATE>` suffix to clearly distinguish Gentoo snapshotted versions form upstream version numbers.

File: sci-electronics/gnucap-models/Manifest [hunk 6]
```diff
@@ -0,0 +1 @@
+DIST gnucap-models-20171003.tar.gz 8048334 BLAKE2B c74a1da922c87dfe78bc3a9249764100221c1ff4ec11a8c1bb2d6f61026dab2fd2cea68c7842973f7239237887eb7825e8d4c97821e6af867ed77240a8206bfb SHA512 40b358554d6bfd579cc388b6c8116bdbf443a11b9b57434f17fc39ddb57f2b5d22fe8847deb4737bd24db004e85503c0bbe639e47df8e293d5c6ed1f08f7f2b8
```
Comment (mgorny) [inline sci-electronics/gnucap-models/Manifest]: I'm sorry but this commit message doesn't conform to our standards. CAT/PN prefix is obligatory.

**Comment by mgorny** [comment]
You still have cross-package commits.

**Comment by Peter-Levine** [comment]
> You still have cross-package commits.

OK. So you want ￼`sci-electronics/gnucap-models` added in its own commit and support added to `sci-electronics/gnucap` in a separate one.

**Comment by mgorny** [comment]
Looks good on my end. @gentoo/sci-electronics, do you confirm? Please note this is adding a new package with you as maintainer.

**Comment by mgorny** [comment]
The package probably needs a real maintainer.

**Comment by Peter-Levine** [comment]
To be clear, I don't actually use these packages.  I just try to resolve bugs in my free time.  If no one merges it, feel free to close.

**Comment by mgorny** [comment]
Ok, something I've just noticed: is this really `0.36.*` or `0.36_pre`, i.e. *before* 0.36 release.

**Comment by felix-salfelder** [comment]
On Wed, Apr 03, 2019 at 12:44:52AM -0700, Michał Górny wrote:
> Ok, something I've just noticed: is this really `0.36.*` or
> `0.36_pre`, i.e. *before* 0.36 release.

good spot. afaics, 0.36 has never been released. some time in 2003 (?)
the version number scheme has switched to yyyymmdd.

0.36_pre* is not wrong, if the order is preserved somehow. in Debian, we
still have 0.36~yyyymmdd (tilde means "pre"), with the intent to drop
the 0.36 at some point -- you might want to consider this...

**Comment by a17r** [comment]
A look at https://repology.org/project/gnucap/versions - I'd say either use `0.36_pre20171003` or switch to `20171003` instead.

**Comment by mgorny** [comment]
I'm closing this due to no activity from submitter.

@felix-salfelder, want to take it over? Feel free to file a new PR, and make yourself the maintainer.

**Comment by felix-salfelder** [comment]
On Sun, Jul 14, 2019 at 12:20:14AM -0700, Michał Górny wrote:
> I'm closing this due to no activity from submitter.
> 
> @felix-salfelder, want to take it over? Feel free to file a new PR, and make yourself the maintainer.

Thanks Michał.

Can't take over the gentoo package. I do maintain the autotools build
system [1] for Gnucap, which might make packaging more straightforward.
maybe next time.

regards
felix

[1] http://gnucap.org/dokuwiki/doku.php/gnucap:manual:autotools

**Pull request closed without merging**